File: lists/majority_element.py

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        result, count = 0, 0
        for i in nums:
            if count == 0:
                result = i
            count += (1 if i == result else -1)
        return result


nums = [1,1,2,2,2,1,1]
res, count = 0, 0
for i in range(len(nums)):
    if count == 0:
        res = nums[i]
        print("i =", i)
        print("nums[i] =", nums[i])
    count += (1 if nums[i] == res else -1)
print("res = ", res)

File: grind75/majority element.cpp
// class Solution {
// public:
//     int majorityElement(vector<int>& nums) {
//         unordered_map <int, int> count_map;
//         int nums_size = nums.size();
//         int maj_ele;
//         for (int i : nums) { // range based for loop
//             count_map[i]++;
//         }
//         // for (auto& [digit, count] : count_map) {
//         //     cout << "digit = " << digit << "count = " << count << "\n";
//         // }
//         // return 0;
//         for (auto& [digit, count] : count_map) {
//             if (count > nums_size/2 ) {
//                 maj_ele = digit;
//             }
//         }
//     return maj_ele;
//     }
// };

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map <int, int> count_map;
        int nums_size = nums.size();
        int maj_ele;
        for (int i : nums) { // range based for loop
            count_map[i]++;
            if (count_map[i] > nums_size/2) {
                maj_ele = i;
            }
        }
    return maj_ele;
    }
};